Formula/rethinkdb.rb

```diff
@@ -20,29 +20,24 @@ class Rethinkdb < Formula
   end
 
   depends_on "boost" => :build
-  depends_on :macos # Due to Python 2 (v8 and gyp fail to build)
-  # https://github.com/Homebrew/linuxbrew-core/pull/19614
-  # https://github.com/rethinkdb/rethinkdb/pull/6401
   depends_on "openssl@1.1"
 
+  uses_from_macos "python" => :build
   uses_from_macos "curl"
-
-  # patch submitted to upstream, https://github.com/rethinkdb/rethinkdb/pull/6934
-  # remove in next release
-  patch do
-    url "https://github.com/chenrui333/rethinkdb/commit/d7d22b4.patch?full_index=1"
-    sha256 "45d387c3360cf6265e27485c273c37224b2ef0000ebd4fdd361dcecc4d2e9791"
-  end
+  uses_from_macos "ncurses"
```

Comment:
Might need to check on ncurses as there isn't any linkage on Linux but there is on macOS.
```
==> brew linkage --cached rethinkdb
System libraries:
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libdl.so.2
  /lib/x86_64-linux-gnu/libgcc_s.so.1
  /lib/x86_64-linux-gnu/libm.so.6
  /lib/x86_64-linux-gnu/libpthread.so.0
  /lib/x86_64-linux-gnu/librt.so.1
  /usr/lib/x86_64-linux-gnu/libstdc++.so.6
Homebrew libraries:
  /home/linuxbrew/.linuxbrew/lib/libcurl.so.4 (curl)
  /home/linuxbrew/.linuxbrew/lib/libcrypto.so.1.1 (openssl@1.1)
  /home/linuxbrew/.linuxbrew/lib/libssl.so.1.1 (openssl@1.1)
  /home/linuxbrew/.linuxbrew/lib/libz.so.1 (zlib)
```